### PR TITLE
Sync current session between phone and watch

### PR DESCRIPTION
**Pull request by by-dam**
- [x] Send start/stop events to the watch

### CODE REVIEW

File: Watch Extension/MRGlanceController.swift [hunk 1]
```diff
@@ -33,7 +35,13 @@ class MRGlanceController: WKInterfaceController, MRSessionProgressRing {
     override func didDeactivate() {
         renderer?.deactivate()
         renderer = nil
+        NSNotificationCenter.defaultCenter().removeObserver(self)
         super.didDeactivate()
     }
+    
+    /// callback function invoked when session is started/ended on the phone
+    internal func sessionDidChange(notif: NSNotification) {
```
Comment (janm399) [inline Watch Extension/MRGlanceController.swift]: Let's spell out notification

File: Watch Extension/MRMainController.swift [hunk 2]
```diff
@@ -126,6 +131,17 @@ class MRMainController: WKInterfaceController, MRSessionProgressRing, MRSessionH
...
         updateUI()
     }
+    
+    /// callback function invoked when session is started/ended on the phone
+    internal func sessionDidStart(notif: NSNotification) {
+        updateUI()
+        renderer?.update()
+    }
+    
+    internal func sessionDidEnd(notif: NSNotification) {
```
Comment (janm399) [inline Watch Extension/MRMainController.swift]: And here

File: Watch Extension/MRMainController.swift [hunk 3]
```diff
@@ -126,6 +131,17 @@ class MRMainController: WKInterfaceController, MRSessionProgressRing, MRSessionH
         MRExtensionDelegate.sharedDelegate().startSession(exerciseType)
         updateUI()
     }
+    
+    /// callback function invoked when session is started/ended on the phone
+    internal func sessionDidStart(notif: NSNotification) {
```
Comment (janm399) [inline Watch Extension/MRMainController.swift]: Same here

File: MuvrKit/MKPhoneExerciseSession+Metadata.swift [hunk 4]
```diff
@@ -0,0 +1,34 @@
+extension MKExerciseSession {
```
Comment (janm399) [inline MuvrKit/MKPhoneExerciseSession+Metadata.swift]: Needs comments & tests

File: MuvrKit/MKPhoneExerciseSession.swift [hunk 5]
```diff
@@ -35,5 +35,13 @@ public struct MKExerciseSession {
         self.completed = exerciseConnectivitySession.last
         self.exerciseType = exerciseConnectivitySession.exerciseType
     }
+    
+    public init(id: String, start: NSDate, end: NSDate?, completed: Bool, exerciseType: MKExerciseType) {
```
Comment (janm399) [inline MuvrKit/MKPhoneExerciseSession.swift]: Needs comments. Is the other initializer still in use?

File: Muvr/MKExerciseSession+MRManagedExerciseSession.swift [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import MuvrKit
+
+extension MKExerciseSession {
```
Comment (janm399) [inline Muvr/MKExerciseSession+MRManagedExerciseSession.swift]: It's a bad idea to construct NSManagedObjects without using the NSEntityDescriptor.... functions. If you do, the CoreData attributes will not work.

File: Muvr/MRAppDelegate.swift [hunk 7]
```diff
@@ -59,7 +59,7 @@ protocol MRApp : MKExercisePropertySource {
     /// - parameter start: the start date, typically value ``NSDate()``
     /// - parameter id: the session identity, typically ``NSUUID().UUIDString``
     ///
-    func startSessionForExerciseType(exerciseType: MKExerciseType, start: NSDate, id: String) throws
+    func startSession(forExerciseType exerciseType: MKExerciseType) throws
```
Comment (janm399) [inline Muvr/MRAppDelegate.swift]: Can we have just one startSession function? This bit in MRAppDelegate is scary (particularly without comments!)

```
    func sessionClassifierDidStart(session: MKExerciseSession) {
        try! startSessionForExerciseType(session.exerciseType, start: session.start, id: session.id)
    }
    
    func startSession(forExerciseType exerciseType: MKExerciseType) throws {
        try startSessionForExerciseType(exerciseType, start: NSDate(), id: NSUUID().UUIDString)
        connectivity.startSession(MKExerciseSession(managedSession: currentSession!))
    }
    
    func startSessionForExerciseType(exerciseType: MKExerciseType, start: NSDate, id: String) throws {
        ...
   }
```

File: Muvr/MKExerciseSession+MRManagedExerciseSession.swift (hunk 6 above)
Comment (by-dam) [inline Muvr/MKExerciseSession+MRManagedExerciseSession.swift]: It doesn't construct a managed object but uses a managed object to construct a ``MKExerciseSession``

**Comment by janm399** [comment]
The whole session starting needs a different structure. The function in the MRApp should not need to supply the sync parameter. And the only thing that the UI can do is to stop the current session.

**Comment by by-dam** [comment]
I would even simplify the start session function to 
```
func startSession(forExerciseType exerciseType: MKExerciseType)
```
as the UI shouldn't have to generate UUID.

(And we're back to the original code with)

**Comment by janm399** [comment]
I think this is good to go now.

**Pull request merged**